src/WalkingModule/src/Module.cpp

```diff
@@ -1497,8 +1531,26 @@ bool WalkingModule::startWalking()
 
 bool WalkingModule::setPlannerInput(const yarp::sig::Vector &plannerInput)
 {
-    m_plannerInput = plannerInput;
+    if (m_motorTemperatureChecker->isThereAMotorOverLimit())
+    {
+        yWarning() << "[WalkingModule::setPlannerInput] The motor temperature is over the limit.";
+        std::vector<unsigned int> indeces = m_motorTemperatureChecker->getMotorsOverLimit();
+        std::string msg = "The following motors temperature are over the limits: ";
+        for (auto index : indeces)
+        {
+            msg += m_robotControlHelper->getAxesList()[index]
+                + ": Max temperature: "
+                + std::to_string(m_motorTemperatureChecker->getMaxTemperature()[index]) + " celsius.";
+        }
+        msg += "The trajectory will be set to zero.";
+        yWarning() << msg;
 
+        m_plannerInput.zero();
+    }
+    else
+    {
+        m_plannerInput = plannerInput;
+    }
```

Comment:
I think we can exploit the "soft stop" logic also for other cases, similar to what @G-Cervettini did as well. Maybe this logic here is a bit too hidden?